fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -2644,6 +2644,57 @@ public CommandProcessingResult forecloseLoan(final Long loanId, final JsonComman
                 .build();
     }
 
+    @Override
+    @Transactional
+    public CommandProcessingResult chargeOff(JsonCommand command) {
+
+        this.loanEventApiJsonValidator.validateChargeOffTransaction(command.json());
+
+        final Map<String, Object> changes = new LinkedHashMap<>();
+        changes.put("transactionDate", command.stringValueOfParameterNamed("transactionDate"));
+        changes.put("locale", command.locale());
+        changes.put("dateFormat", command.dateFormat());
+
+        // TODO: add business logic validation (transaction date cannot be future, cannot be earlier than latest
```

Comment:
Don't we wanna do this in this PR? Or is there a follow up ticket for this?

Reply:
Follow up tickets... should i remove them?